sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,423 @@
+    >>> from sktime.classification.interval_based import TimeSeriesForestClassifier
+    >>> from sktime.datasets import load_unit_test
+    >>> X_train, y_train = load_unit_test(split="train", return_X_y=True)
+    >>> X_test, y_test = load_unit_test(split="test", return_X_y=True)
+    >>> clf = TEASER(
+    ...     classification_points=[6, 16, 24],
+    ...     estimator=TimeSeriesForestClassifier(n_estimators=10)
+    ... )
+    >>> clf.fit(X_train, y_train)
+    TEASER(...)
+    >>> y_pred = clf.predict(X_test)
+    """
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:multithreading": True,
+        "capability:early_prediction": True,
+    }
+
+    def __init__(
+        self,
+        estimator=None,
+        classification_points=None,
+        n_jobs=1,
+        random_state=None,
+    ):
+        self.estimator = estimator
+        self.classification_points = classification_points
+
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+
+        self._estimators = []
+        self._classification_points = []
+        self._svms = []
+        self._consecutive_predictions = 0
+
+        self._svm_gammas = [100, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1.5, 1]
+
+        super(TEASER, self).__init__()
+
+    def _fit(self, X, y):
+        n_instances, _, series_length = X.shape
+
+        self._classification_points = (
+            copy.deepcopy(self.classification_points)
+            if self.classification_points is not None
+            else [round(series_length / i) for i in range(1, 21)]
+        )
+        # remove duplicates
+        self._classification_points = list(set(self._classification_points))
+        self._classification_points.sort()
+        # remove classification points that are less than 3
+        self._classification_points = [i for i in self._classification_points if i >= 3]
+        # make sure the full series length is included
+        if self._classification_points[-1] != series_length:
+            self._classification_points.append(series_length)
+        # create dictionary of classification point indicies
+        self._classification_point_dictionary = {}
+        for index, classification_point in enumerate(self._classification_points):
+            self._classification_point_dictionary[classification_point] = index
+
+        m = getattr(self.estimator, "n_jobs", None)
+        threads = self._threads_to_use if m is None else 1
+
+        fit = Parallel(n_jobs=threads)(
+            delayed(self._fit_estimator)(
+                X,
+                y,
+                i,
+            )
+            for i in range(len(self._classification_points))
+        )
+
+        self._estimators, self._svms, X_svm, train_preds = zip(*fit)
+
+        # tune consecutive predictions required to best harmonic mean
+        best_hm = -1
+        for g in range(2, min(6, len(self._classification_points))):
+            state_info = [(0, 0, 0) for _ in range(n_instances)]
```

Comment:
Not sure, if I understand the state_info correctly. Maybe encapsulate it into some methods and add comments?

I personally would have used lists of dicts() - not claiming this works better

Reply:
It's mostly modelled after the state_info from decide, but I could split it up. Need to store the current time stamp used for earliness, the number of consecutive decisions made and the class prediction. 

Will add comments explaining it a bit more for now.

Reply:
Same as above :)